### PR TITLE
fix: add css rules to devtools extension debugger when pass renderer on SSR

### PR DESCRIPTION
**Pull request by Dwlad90**
This PR fixes https://github.com/microsoft/griffel/issues/173.

### CODE REVIEW

File: packages/core/src/renderer/createDOMRenderer.ts [hunk 1]
```diff
@@ -67,14 +67,17 @@ export function createDOMRenderer(
             metadata,
           );
 
+          if (process.env.NODE_ENV !== 'production' && isDevToolsEnabled) {
+            if (!debugData.hasCSSRule(ruleCSS)) {
+              debugData.addCSSRule(ruleCSS);
+            }
+          }
+
```
Comment (layershifter) [inline packages/core/src/renderer/createDOMRenderer.ts:75]: As it's SSR specific I was wondering if we could fill data in `rehydrateRendererCache()` instead:

https://github.com/microsoft/griffel/blob/79c97b9ec6b4b028e31e17b613e6d89a37bfd6a7/packages/core/src/renderer/rehydrateRendererCache.ts#L41-L45

Will it work?

File: packages/core/src/devtools/store.ts [hunk 2]
```diff
@@ -41,4 +41,7 @@ export const debugData = {
   getSequenceDetails: (sequenceHash: SequenceHash): typeof sequenceDetails[string] | undefined => {
     return sequenceDetails[sequenceHash];
   },
+  hasCSSRule: (rule: string): boolean => {
+    return cssRules.includes(rule);
+  },
```
Comment (layershifter) [inline packages/core/src/devtools/store.ts:46]: I am thinking to turn `cssRules` into `Set` if it's really needed:

```diff
-const cssRules: string[] = [];
+const cssRules = new Set<string>();
```

```diff
  addCSSRule: (rule: string) => {
-    cssRules.push(rule);
+    cssRules.add(rule);
  },
```

```diff
  getCSSRules: (): string[] => {
-   return cssRules;
+   return Array.from(cssRules);
  },
```

**Comment by layershifter** [comment]
FYI @YuanboXue-Amber

**Comment by layershifter** [comment]
@Dwlad90 thanks for contributing, I left few comments 🙂

File: packages/devtools/src/utils.ts [hunk 3]
```diff
@@ -10,12 +10,14 @@ export function getRulesBySlots(node: DebugResult, result: SlotInfo[] = []): Slo
...
         sourceURL,
-        rules: debugClassNames.map(({ className, overriddenBy }) => {
-          return {
-            cssRule: rules![className],
-            overriddenBy,
-          };
-        }),
+        rules: debugClassNames
+          .filter(({ className }) => !!rules?.[className])
+          .map(({ className, overriddenBy }) => {
```
Comment (Dwlad90) [inline packages/devtools/src/utils.ts:15]: Use reduce instead of filter and map

File: packages/core/src/devtools/store.ts (hunk 2 above)
Comment (Dwlad90) [reply packages/core/src/devtools/store.ts:46]: Yes, this will solve the duplication problems and eliminate the need to check if the string exists.

**Comment by Dwlad90** [comment]
> 

@layershifter Thanks for a code review, I've update the PR according to your suggestions

**Review by YuanboXue-Amber — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Pull request merged**